README.md

```diff
@@ -2,154 +2,162 @@
-3. [Filecoin Pin dApp Demo](https://github.com/filecoin-project/filecoin-pin-website) - Run or fork a simple demo dApp using Filecoin Pin
+Filecoin Pin is a fully decentralized persistence layer for IPFS content using the global network of Filecoin storage providers with cryptographic guarantees.
 
-All methods use the [Synapse SDK](https://synapse.filecoin.services/) to create Filecoin storage deals, providing persistent storage with cryptographic proofs.
+When you use Filecoin Pin, your IPFS files gain:
 
-## Installation
+- **Verifiable persistence** - Storage providers must cryptographically prove daily that they continue to store and serve your data
+- **Economic incentives** - You only pay when storage proofs are successfully delivered and verified onchain
+- **Decentralized infrastructure** - Your data can be stored across a global network of independent storage providers
+- **Seamless IPFS integration** - Continue using standard [IPFS Mainnet](https://docs.ipfs.tech/concepts/glossary/#mainnet) tooling (e.g., Kubo, Helia, HTTP Gateways) while gaining Filecoin's persistence guarantees
+- **Sovereign data** - Choose your providers, audit storage proofs and payments onchain, with no dependency on a single company
 
-Requires Node.js 24+
+## Who is Filecoin Pin for?
 
-```bash
-npm install -g filecoin-pin
-```
+Filecoin Pin is designed for **developers building on IPFS** who need trustless, economically-incentivized persistence for their content. Whether you're building dApps, CI/CD workflows, static websites, AI agents, or other applications, Filecoin Pin provides the missing persistence layer for IPFS.
 
-## Quick Start
+## Affordances
 
-### 1. Set Up Payments (Required First Step)
+Filecoin Pin offers multiple "affordances" to integrate Filecoin storage into your workflow:
 
-Before storing data, configure your Filecoin payment approvals:
+### CLI
+Upload IPFS files directly to Filecoin via the command line. Perfect for developers who want to integrate Filecoin storage into scripts, workflows, or local development environments.
 
-```bash
-# Check your current payment status
-filecoin-pin payments status
+- **Repository**: This repo ([filecoin-project/filecoin-pin](https://github.com/filecoin-project/filecoin-pin))
+- **Documentation**: 
+  - Run `filecoin-pin --help` to see all available commands and options.
+  - [CLI Walkthrough](https://docs.filecoin.io/builder-cookbook/filecoin-pin/filecoin-pin-cli)
+- **Installation**: `npm install -g filecoin-pin`
 
-# Interactive setup (recommended)
-filecoin-pin payments setup
+### GitHub Action
+Automatically publish websites or build artifacts to IPFS and Filecoin as part of your CI/CD pipeline. Ideal for static websites, documentation sites, and automated deployment workflows.
 
-# Or automated setup
-filecoin-pin payments setup --deposit 100 --storage 10TiB --auto
-```
+- **Repository**: This repo ([upload-action/](./upload-action))
```

Comment:
```suggestion
- **Repository**: This repo ([see upload-action/](./upload-action))
```

matches the one below I think, the one above just has the repo name, this one doesn't, it reads a little odd